tests/functional/api/test_groups.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+import base64
+
+
+from h.models.auth_client import GrantType
+
+
+@pytest.mark.functional
+class TestCreateGroup(object):
+
+    def test_it_returns_http_200_with_valid_payload(self, app, token_auth_header):
+        group = {
+            'name': 'My Group'
+        }
+
+        res = app.post_json('/api/groups', group, headers=token_auth_header)
+
+        assert res.status_code == 200
+
+    def test_it_returns_http_400_with_invalid_payload(self, app, token_auth_header):
+        group = {}
+
+        res = app.post_json('/api/groups', group, headers=token_auth_header, expect_errors=True)
+
+        assert res.status_code == 400
+
+
+@pytest.mark.functional
+class TestAddMember(object):
+
+    def test_it_returns_http_204_when_successful(self, app, user, group, auth_client_header):
+        res = app.post_json("/api/groups/{pubid}/members/{userid}".format(pubid=group.pubid, userid=user.userid),
+                            headers=auth_client_header)
```

Comment:
I think it might be possible to use Pyramid URL generation ([`route_url()`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.route_url), [`route_path()`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.route_path), [`resource_url()`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.resource_url), [`resource_path()`](https://docs.pylonsproject.org/projects/pyramid/en/latest/api/request.html#pyramid.request.Request.resource_path)), that might make the tests a little less brittle by avoiding hard-coding URL paths into them, but it would require a Pyramid request object configured with h's routes and resources. We could probably provide a `route_path()` fixture to the tests.

Reply:
I see several spots in the functional tests where reusable fixtures and better fixtures would be nice. I think I'm going to get a bunch of tests written and _then_ optimize them. But, well spotted! Thanks!